cluster/ci/origin/pipelines/deploy/Jenkinsfile

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent any
+  stages {
+    stage("Configure Variables") {
+      steps {
+        script {
+          this._buildName = "pr-17205-1efbe34c40a136308ba66b96"
+        }
+      }
+    }
+    stage("Extract Config Vars") {
+      steps {
+        script {
+          openshift.withCluster() {
+            def cm = openshift.selector("cm/${this._buildName}").object()
+            this._imageBase = cm.data["image-base"]
+            this._rpmRepo = cm.data["rpm-repo"]
+          }
+        }
+      }
+    }
+    stage("Extract secrets") {
+      steps {
+        script {
+          openshift.withCluster() {
+            def secret = openshift.selector("secret/gce-provisioner").object()
+            def secretBytes = secret.data["gce.json"].decodeBase64()
+            def secretData = new String(secretBytes)
+            dir("cluster/test-deploy/data") {
+              writeFile file: "gce.json", text: secretData
+            }
+
+            secret = openshift.selector("secret/gce-ssh").object()
+            secretBytes = secret.data["ssh-privatekey"].decodeBase64()
+            def secretBytesPub = secret.data["ssh-publickey"].decodeBase64()
+            secretData = new String(secretBytes)
+            def secretDataPub = new String(secretBytesPub)
+            dir("cluster/test-deploy/data") {
+              writeFile file: "ssh-privatekey", text: secretData 
+              writeFile file: "ssh-publickey", text: secretDataPub
+	    }
+          }
+          this._token = sh(script: "oc whoami -t", returnStdout: true)
+          dir('cluster/test-deploy/data') {
+            stash includes: '*', name: 'deploy-data'
+          }
+        }
+      }
+    }
+    stage("Run Deployment") {
+      agent {
+        kubernetes {
+          cloud 'openshift'
+          label 'orgin-gce'
+          containerTemplate {
+            name 'origin-gce'
+            image 'openshift/origin-gce:latest'
+            ttyEnabled true
+            command 'cat'
+          }
+        }
+      }
+      steps {
+	container('origin-gce') {
+	  script {
+	    sh "mkdir ansible-files"
+            dir("ansible-files") {
+            	unstash 'deploy-data'
```

Comment:
Hmmm feels like we should just mount the secrets into the `agent` instead of this